### DISCUSSION TITLE
挑戰題、how to get twitch  OAuth token

### DISCUSSION THREAD

**Discussion by s103071049** [Lidemy/forum#91] [其他問題] [answered]
嗨以，大家好，不好意思又打擾大家了~感謝大家的幫忙
"error":"Unauthorized","status":401,"message":"Invalid OAuth token"
* 參考資料 : https://dev.twitch.tv/docs/api/reference#get-streams

我要怎麼拿到 Authorization 
我最初一直以為，是將下面的這些放到 headers 去，就像前面的作業一樣。 但不管怎麼做，{"error":"Unauthorized","status":401,"message":"Invalid OAuth token"}。所以應該是 "authorized" 。

``` js
// 參考資料，想提取的 url
curl -X GET 'https://api.twitch.tv/helix/streams' \
-H 'Authorization: Bearer 2gbdx6oar67tqtcmt49t3wpcgycthx' \
-H 'Client-Id: wbmytr93xzw8zbg0p1izqyzzc5mbiz'
```

``` js
const request = require('request')
const process = require('process')

request(
  oprions = {
    url: `https://api.twitch.tv/helix/streams`,
    headers: {
    	'Client-ID': 'ka1mgyrfu09rtyw1znuocyf6x0xciq',
    	'Accept': 'application/vnd.twitchtv.v5+json',
    	'Authorization': ' Bearer 2gbdx6oar67tqtcmt49t3wpcgycthx'
    }
  },

  (error, response, body) => {
    const json = JSON.parse(body)
    console.log(body)
    console.log(json)
  }

) 

```

試圖解決作法 :
https://dev.twitch.tv/docs/authentication/getting-tokens-oauth，但這邊好像沒有我要的東西。

**Comment by allenliaooooooo** [comment, marked as answer]
你這組 token 應該是 Twitch 的 example 放的那組吧？
應該要想辦法拿到一組自己的 token
然後你的 token 前面有一個空格，不確定會不會影響

你的試圖解決做法是對的～

**Reply by s103071049** [reply]
我成功的抓出來了範例的結果，只是抓出來還沒有處理 limit 的問題~

以下是實作過程 :

首先、我不知道怎麼選token我看不懂官方給的表，所以作法是 google 看看有沒有類似網友分享的然後他使用哪個 token，這個方法滿爛的也不一定保證他找的是我要的，總之就用了 OAuth client credentials flow。然後在剛剛從他的官方文件中，type of token 了解為甚麼要是使用這個 token (reason : Authenticate your app and allow it to access resources that it owns)。所以應該先去理解名詞再 google ，會節省時間。
* 熱心網友的分享資料 : https://ovvo.cc/vue-twitch-api/
* client secret 可以在主控台找到、scope這個參數他的範例也沒放也ok ( Without specifying scopes, your app can access only basic information about the authenticated user. 另外我想要的url 並沒有指定要放scope，所以我就沒放他了)

接著是整個問題卡最久的部分，我一直以為他給的 example 就是 code ，但準確地說是類似虛擬碼的東西。所以必須將這些轉成程式碼。

然後按照他給的一步一步console.log 出來，我也不確定會不會跑出來。不對的話，就調空格、大小寫盡可能和他長得一樣，然後再看看結果。

結論 : 我覺得閱讀官方文件並不是一件容易的事。不容易的點在1)你不知道在那裡可以回答你的問題 2)對方表達的和你想的不是同一件事情。
感謝大家的幫忙~

**Reply by ddylanlinn** [reply]
深有同感😥，謝謝你紀錄的滿有條理的

**Reply by s103071049** [reply]
一起加油!!!
